src/test/scala/com/github/simplesteph/ksm/parser/YamlAclParserTest.scala

```diff
@@ -0,0 +1,170 @@
+package com.github.simplesteph.ksm.parser
+
+import java.io.StringReader
+
+import kafka.security.auth._
+import org.apache.kafka.common.resource.PatternType
+import org.apache.kafka.common.utils.SecurityUtils
+import org.scalatest.{FlatSpec, Matchers}
+
+class YamlAclParserTest extends FlatSpec with Matchers {
+
+  val row = Map(
+    "KafkaPrincipal" -> "User:alice",
+    "ResourceType" -> "Topic",
+    "PatternType" -> "LITERAL",
+    "ResourceName" -> "test",
+    "Operation" -> "Read",
+    "PermissionType" -> "Allow",
+    "Host" -> "*",
+  )
+
+  val resource = Resource(Topic, "test", PatternType.LITERAL)
+  val acl = Acl(SecurityUtils.parseKafkaPrincipal("User:alice"), Allow, "*", Read)
+  val yamlAclParser = new YamlAclParser()
+
+  "aclsFromYaml" should "correctly parse a Correct YAML" in {
+    val yaml =
+      """#Ignore comments
+        |users:
+        |  alice: # Ignore comments
+        |    groups:
+        |      mygroup-*:
+        |        - Write,Deny,12.34.56.78
+        |    topics:
+        |      topic1: [ Consume, Produce ]
+        |  bob:
+        |    groups:
+        |      group1: [ Read, Write ]
+        |      group2:
+        |        - Describe,Allow,*
+        |        - Read
+        |  peter:
+        |    transactional_ids:
+        |      'tr1*':
+        |        - All
+        |    clusters:
+        |      '*':
+        |        - Create,Allow,*
+        |  tom:
+        |    topics:
+        |      '*':
+        |        - Admin
+        |""".stripMargin
+
+    val res = yamlAclParser.aclsFromReader(new StringReader(yaml))
+
+    res.acls.map(_.toString()) shouldBe Set(
```

Comment:
this test relies on `.toString` which I'm not a fan of, as it's most likely bound to change with new versions of Kafka. Can we instead check the results of `res.acls` ? It's a bit more annoying to write as you have to construct ACLs but it's a lot more stable

Reply:
Yeah I will rewrite the whole test suite, it was a first iteration to get them decent based on the new API